### ISSUE TITLE
group does not work for subcommand?

### ISSUE THREAD

**Issue by Geobert** [TeXitoi/structopt#122]
Hi,
```rust
fn input_arg_group() -> ArgGroup<'static> {
    ArgGroup::with_name("input").multiple(false);
    ArgGroup::with_name("input").required(true)
}

arg_enum!{
#[derive(Debug)]
pub enum Command {
    Time,
}
}

#[derive(StructOpt, Debug)]
#[structopt(name = "clicom", raw(group = "input_arg_group()"))]
pub struct Opt {
    /// Data stream to send to the device (if none, --file must be used)
    #[structopt(group = "input")]
    pub data: Option<HexData>,

    /// Script file to play (if none, `data` must be provided)
    #[structopt(subcommand, group="input")]
    pub script: Option<ScriptSub>,

    /// Buit-in commands
    #[structopt(
        short = "c",
        long = "command",
        group = "input",
        raw(possible_values = "&Command::variants()", case_insensitive = "true")
    )]
    pub command: Option<Command>,

    /// Target device serial number
    // #[structopt(short = "s", long = "serial")]
    // pub device_serial: Option<String>,

    /// Run the command in a forever loop, Ctrl-C to stop
    #[structopt(long = "loop")]
    pub in_loop: bool,

    /// Activate debug mode
    #[structopt(short = "d", long = "debug")]
    pub debug: bool,
}

#[derive(StructOpt, Debug)]
pub enum ScriptSub {
    #[structopt(name = "script")]
    Script {
        #[structopt(short = "v", long = "verbose")]
        verbose: bool,
        #[structopt(parse(from_os_str))]
        script_file: PathBuf,
    },
}
```

I got:
```
error: proc-macro derive panicked
  --> src\cli_args.rs:18:10
   |
18 | #[derive(StructOpt, Debug)]
   |          ^^^^^^^^^
   |
   = help: message: methods in attributes is not allowed for subcommand
```

Is there a way to make a subcommand part of a group? (I want to make sure that one and only one of `data`, `script` or `command` is present)

**Comment by TeXitoi** [comment]
I don't understand what your trying to do. Please give commands with if it should be accepted or refused as you would type on the terminal.

**Comment by Geobert** [comment]
`clicom` -> rejected
`clicom 9009` -> accepted
`clicom -c a_command` accepted
`clicom script file_name` accepted
`clicom script file_name --verbose` accepted
`clicom 9009 -c a_command` rejected
`clicom 5000 --verbose` rejected

**Comment by TeXitoi** [comment]
I can't find any way to do it using clap. If that's not possible with clap, then that's not possible with structopt. You can still check by hand after successful parsing.

**Comment by Geobert** [comment]
Ok I understand, I just wanted to be sure I was not missing something :)

**Comment by TeXitoi** [comment]
From https://docs.rs/clap/2.32.0/clap/struct.ArgGroup.html

> ArgGroups are a family of related arguments [...]

Thus I don't think group can be used on subcommands.

**Comment by keturn** [comment]
I think #151 was a duplicate of this, and is now fixed in v0.3.6.

**Comment by CreepySkeleton** [comment]
And 0.3.6 was just yanked 😱 . We'll work it out, don't you worry.
